rholang/src/test/scala/coop/rchain/rholang/interpreter/SubstituteTest.scala

```diff
@@ -0,0 +1,166 @@
+package coop.rchain.rholang.interpreter
+
+import org.scalatest.{FlatSpec, Matchers}
+import Substitute._
+import scala.collection.immutable.HashMap
+import ExprTest._
+
+object ExprTest {
+  val chan0: Channel = ChanVar(BoundVar(0))
+  val name0: Quote = Quote(Par.forge)
+  val chan1: Channel = ChanVar(BoundVar(1))
+  val name1: Quote = Quote(Par.forge)
+  val chan2: Channel = ChanVar(BoundVar(2))
+  val name2: Quote = Quote(Par.forge)
+  val send0: Send = Send(chan0, List(Par()),false,0)
+  val send1: Send = Send(chan1, List(Par()),false,0)
+  val send2: Send = Send(chan2, List(Par()),false,0)
+  val sendQ: Send = Send(Quote(send0),List(Par()),false,0)
+}
+
+class VarSub extends FlatSpec with Matchers {
+
+  "FreeVar" should "throw an error" in {
+    val env = HashMap(0 -> Par.forge)
+    an [Error] should be thrownBy subOrDec(FreeVar(0))(env)
+  }
+
+  "BoundVar" should "be substituted for process" in {
+    val source = Par.forge
+    val env = HashMap(0 -> source)
+    val result = subOrDec(BoundVar(0))(env)
+    result should be (Right(source))
+  }
+
+  // BoundVar(0), Env := { 0 -> Par(BoundVar(0)) --> Par(BoundVar(1)), Env := { 0 -> Par(BoundVar(0)) }
+  "BoundVar" should "be substituted with renamed expression" in {
+    val source = Par(send0)
+    val _source = Par(send1)
+    val env = HashMap(0 -> source)
+    val result = subOrDec(BoundVar(0))(env)
+    result should be (Right(_source))
+  }
+
+  "BoundVar" should "throw an error" in {
+    val env = HashMap(1 -> Par.forge)
+    an [Error] should be thrownBy subOrDec(BoundVar(0))(env)
+  }
+
+  "BoundVar" should "be decremented by environment level" in {
+    val env = HashMap(0 -> Par.forge, 1 -> Par.forge)
+    val result = subOrDec(BoundVar(2))(env)
+    result should be (Left(BoundVar(0)))
+  }
+}
+
+class ChannelSub extends FlatSpec with Matchers {
+
+  "ChanVar" should "be decremented by environment level" in {
+    val env = HashMap(0 -> Par.forge, 1 -> Par.forge)
+    val result = substitute(chan2)(env)
+    result should be (chan0)
```

Comment:
Please define these closer to the tests that use them.

Reply:
They're used globally.